### PR TITLE
Use string_view for Settings key.

### PR DESCRIPTION
**Pull request by vng**

### CODE REVIEW

File: base/base_tests/stl_helpers_tests.cpp [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 
 #include <algorithm>
 #include <deque>
-#include <utility>
+#include <map>
 #include <vector>
 
-using namespace base;
 
 namespace stl_helpers_test
```
Comment (biodranik) [inline base/base_tests/stl_helpers_tests.cpp:11]: nit:
```suggestion
namespace stl_helpers_tests
```

File: map/framework.cpp [hunk 2]
```diff
@@ -88,23 +88,24 @@ Framework::FixedPosition::FixedPosition()
...
-char const kAllowAutoZoom[] = "AutoZoom";
-char const kTrafficEnabledKey[] = "TrafficEnabled";
-char const kTransitSchemeEnabledKey[] = "TransitSchemeEnabled";
-char const kIsolinesEnabledKey[] = "IsolinesEnabled";
-char const kOutdoorsEnabledKey[] = "OutdoorsEnabled";
-char const kTrafficSimplifiedColorsKey[] = "TrafficSimplifiedColors";
-char const kLargeFontsSize[] = "LargeFontsSize";
-char const kTranslitMode[] = "TransliterationMode";
-char const kPreferredGraphicsAPI[] = "PreferredGraphicsAPI";
-char const kShowDebugInfo[] = "DebugInfo";
```
Comment (biodranik) [inline map/framework.cpp:103]: Curious, what was wrong with char arrays? )

File: platform/settings.hpp [hunk 3]
```diff
@@ -33,22 +31,22 @@ class StringStorage : public platform::StringStorageBase
 /// Retrieve setting
 /// @return false if setting is absent
 template <class Value>
-[[nodiscard]] bool Get(std::string const & key, Value & outValue)
+[[nodiscard]] bool Get(std::string_view key, Value & outValue)
```
Comment (biodranik) [inline platform/settings.hpp:34]: nit: std::optional is a good candidate for this case.

File: platform/string_storage_base.hpp [hunk 4]
```diff
@@ -12,13 +12,13 @@ class StringStorageBase
...
+  bool GetValue(std::string_view key, std::string & outValue) const;
+  void SetValue(std::string_view key, std::string && value);
   void Update(std::map<std::string, std::string> const & values);
-  void DeleteKeyAndValue(std::string const & key);
-  
+  void DeleteKeyAndValue(std::string_view key);
+
 private:
-  using Container = std::map<std::string, std::string>;
+  using Container = std::map<std::string, std::string, std::less<>>;
```
Comment (biodranik) [inline platform/string_storage_base.hpp:21]: Why explicit less? Isn't it already defined?

**Review by biodranik — APPROVED** [review]

File: map/framework.cpp (hunk 2 above)
Comment (vng) [reply map/framework.cpp:103]: Initialize string_view once (calculate length).

File: platform/settings.hpp (hunk 3 above)
Comment (vng) [reply platform/settings.hpp:34]: Nah, std::string_view then or std::string const *

File: platform/string_storage_base.hpp (hunk 4 above)
Comment (vng) [reply platform/string_storage_base.hpp:21]: is_transparent find

File: platform/settings.hpp (hunk 3 above)
Comment (vng) [reply platform/settings.hpp:34]: - ```optional<string>``` is a strange pattern itself (we don't need an empty setting)
- avoid useless copy

**Pull request merged**